### PR TITLE
Add known variables; add PKGSEC check

### PR DESCRIPTION
**Pull request by Cyanoxygen**
- Known variables is hard coded into the project, so it might need to be updated once autobuild is changed.
- List of valid PKGSEC variables is also hard-coded, but if autobuild is installed on the host system, it will try to read that file.

### CODE REVIEW

**Comment by Cyanoxygen** [comment]
Previously:

image

Now undefined variable error for `PKGDIR`, `SRCDIR`, etc. are gone:

image

**Review by eatradish — DISMISSED** [review]

**Comment by Cyanoxygen** [comment]
`PKGREL` should not be parsed as an integer. Fixing up.

File: apml/src/apml/variables.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"ASFLAGS",
+	"CPPFLAGS",
+	"LDFLAGS",
+	"RUSTFLAGS",
+	// Various build-time variables
+	"ABMK",
+];
+
+/// Returns `true` if the given string is in the known variables list.
+pub fn is_known_variable(v: String) -> bool {
```
Comment (eatradish) [inline apml/src/apml/variables.rs]: Use `&str`

File: apml/src/apml/variables.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/// variables.rs - Known variables to skip the undefined variable error.
+
+/// Known variables during the execution of autobuild.
+///
+/// Undefined variable errors should be ignored for these variables.
+///
+/// Part of this collection is extracted from [`autobuild4/lib/default-paths.sh`](file:///usr/lib/autobuild4/lib/default-paths.sh)
+const KNOWN_VARIABLES: &[&str] = &[
```
Comment (eatradish) [inline apml/src/apml/variables.rs:8]: Maybe we can use `HashSet` ?

**Review by eatradish — CHANGES_REQUESTED** [review]

**Pull request merged**